ciao-controller/internal/datastore/datastore.go

```diff
@@ -2280,3 +2257,61 @@ func (ds *Datastore) UnMapExternalIP(address string) error {
 
 	return nil
 }
+
+func (ds *Datastore) generateCNCIWorkload() {
+	// generate the CNCI workload.
+	config := `---
+#cloud-config
+users:
+  - name: cloud-admin
+    gecos: CIAO Cloud Admin
+    lock-passwd: false
+    passwd: $6$rounds=4096$w9I3hR4g/hu$AnYjaC2DfznbPSG3vxsgtgAS4mJwWBkcR74Y/KHNB5OsfAlA4gpU5j6CHWMOkkt9j.9d7OYJXJ4icXHzKXTAO.
+    sudo: ALL=(ALL) NOPASSWD:ALL
+...
+`
+	cpus := payloads.RequestedResource{
+		Type:      payloads.VCPUs,
+		Value:     4,
+		Mandatory: false,
+	}
+
+	mem := payloads.RequestedResource{
+		Type:      payloads.MemMB,
+		Value:     128,
```

Comment:
@kaccardi the 128MB CNCI cannot handle full 10 Gbps worth of traffic, which is our minimal cluster requirement. So as @markdryan suggests that is a parameter we would like to control. However that should be derived based on the size of the tenant (vs an explicit CNCI configuration IMO). I know we do not have a tenant size hint today.